servlet/src/main/java/io/undertow/servlet/spec/HttpServletRequestImpl.java

```diff
@@ -374,6 +375,12 @@ public String changeSessionId() {
         if (session == null) {
             throw UndertowServletMessages.MESSAGES.noSession();
         }
+        if (this.exchange.getAttachment(ServletRequestContext.ATTACHMENT_KEY).getServletResponse().isCommitted()) {
+            if (!this.servletContext.getDeployment().getDeploymentInfo().isOrphanSessionAllowed()) {
+                throw UndertowServletMessages.MESSAGES.sessionIdChangeAfterResponseCommittedNotAllowed();
+            }
+            UndertowServletLogger.REQUEST_LOGGER.sessionIdChangeAfterResponseCommitted();
```

Comment:
This log is still logged at WARN level even if allowOrphanSession is set to true. So, if a user's application is not compliant with the servlet spec, they can start to see either many IllegalStateException or many WARN log messages after migrating to a newer version. 
- allowOrphanSession == false -> IllegalStateException is thrown.
- allowOrphanSession == true -> WARN message is logged.

I think this message should be logged at DEBUG level (or logged at WARN level only once) if allowOrphanSession == true.

Reply:
@msfm Worst case scenario, the log frequency would be once per completed user session.
That said, given that an application would have explicitly enabled this, I agree, it is fair to log at a lower level.